File: Hash_Table/454.4Sum_II/4Sum_II.py
class Solution:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        ans = 0
        hash_map = defaultdict(int)
        for i in nums1:
            for j in nums2:
                hash_map[i + j] += 1
        
        for k in nums3:
            for l in nums4:
                flag = hash_map[-1 * (k + l)]
                if flag: ans += flag
        return ans

File: Hash_Table/454.4Sum_II/4Sum_II.cpp
class Solution {
public:
    int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
        int ans = 0;
        unordered_map<int,int> hash_map;
        for(int i: nums1){
            for(int j: nums2){
                hash_map[i + j]++;
            }
        }
        for(int k: nums3){
            for(int l: nums4){
                int flag = hash_map[-1 * (k + l)];
                if(flag) ans += flag;
            }
        }
        return ans;
    }
};